src/index/txindex.cpp

```diff
@@ -79,7 +79,7 @@ bool TxIndex::FindTx(const uint256& tx_hash, uint256& block_hash, CTransactionRe
         return false;
     }
 
-    AutoFile file{m_chainstate->m_blockman.OpenBlockFile(postx, true)};
+    AutoFile file{m_blockman.OpenBlockFile(postx, true)};
```

Comment:
I feel like the entire `FindTx` function could just be moved into the block manager. Then there is no need to pass it around here and it can instead be called through the interface just like elsewhere.